### ISSUE TITLE
Allow subjects to be used for non-copyable items

### ISSUE THREAD

**Issue by helgoboss** [rxRust/rxRust#74]
Currently subjects only work for copyable items or references (please correct me if I'm wrong or if there's a workaround). I wish there would be a way to support items which are not copyable but implement `Clone`. Or at least a way to use items wrapped in shared pointers (`Rc`).

I would be happy to make an implementation attempt once the trait refactoring (#62) has progressed.

Background: I have a struct which I would like to use as observable item. However, it's impossible to implement `Copy` because it owns a `String`. One workaround would be to use references but then I have to deal with lifetime issues which I don't want and is also unnecessary in my case. So I was thinking of wrapping the item in an `Rc` (cloning `String` over and over again wouldn't be the best idea anyway) ... but `Rc` is also not copyable.

**Comment by helgoboss** [comment]
@M-Adoo Is there any specific reason you have the `Copy` requirement? My guess is you want to prevent users from building observable chains that have bad performance (which would happen if they use items which are expensive to clone) ... is that correct?

**Comment by M-Adoo** [comment]
Yes，just performance consider. But maybe `Copy` too strict. I'm focusing on #73 now, after #73 we can try to do something for this issue.

**Comment by helgoboss** [comment]
@M-Adoo So the easiest thing would be to just replace each `Copy` with a `Clone` in `subject.rs` and `observer.rs` and call `clone()` in `observer.rs` when calling `next` or `error`. Of course then it's up to the user to not do anything stupid performance-wise. Would you be okay to receive this as PR or do you have another idea?

**Comment by M-Adoo** [comment]
Maybe we can define a `SubjectCopy` replace `Copy`,  and implement it for any type which implemented `Copy`.  User can implement `SubjectCopy` for its type. How about this way ?

**Comment by helgoboss** [comment]
That sounds good and is more explicit. Then I think rxRust should implement `SubjectCopy` also for common non-copyable but cheap-to-clone wrapper types like `Rc<T>`, `Cell<T>`, `RefCell<T>` and `Arc<T>` where `T: SubjectCopy` or `T: Copy`. Agree?

I wonder if we should choose a more general name than `SubjectCopy`, such as `PayloadCopy`. Because for me it looks like this change doesn't affect just subjects, does it? In `observer.rs` there's also a bound demanding the `Copy` trait:

```rust
impl<Item, Err, T> Observer<Item, Err> for Vec<T>
where
  T: Publisher<Item, Err>,
  Item: Copy,
  Err: Copy,
```

Is this observer implementation used by subject logic only?

**Comment by M-Adoo** [comment]
> That sounds good and is more explicit. Then I think rxRust should implement SubjectCopy also for common non-copyable but cheap-to-clone wrapper types like Rc<T>, Cell<T>, RefCell<T> and Arc<T> where T: SubjectCopy or T: Copy. Agree?

I agree.

> I wonder if we should choose a more general name than SubjectCopy, such as PayloadCopy. 
Yes.  `SubjectCopy` is not  my thoughtful name,  I'm glad you give a better.

**Comment by M-Adoo** [comment]
closed by #83
